### PR TITLE
Add Toolpad file schema reference to the docs

### PR DESCRIPTION
**Pull request by Janpot**
Generates documentation that describes the Toolpad configuration files

See https://deploy-preview-1940--mui-toolpad-docs.netlify.app/toolpad/schema-reference/

### CODE REVIEW

File: docs/schemas/v1/definitions.json [hunk 1]
```diff
@@ -0,0 +1,616 @@
...
+                      "required": [
+                        "name"
+                      ],
+                      "additionalProperties": false,
+                      "description": "A name/value pair."
+                    },
+                    "description": "Parameters to pass to this query."
+                  },
+                  "mode": {
+                    "type": "string",
```
Comment (mnajdova) [inline docs/schemas/v1/definitions.json:83]: Not related to this PR: This can probably be better typed.

**Review by mnajdova — COMMENTED** [review]
For the types that are described below, like `NameStringValuePair`, would it be feasable to show them in a tooltip when hovering? It will help with not having to scroll back and forth. 
image

Looks much better than the first iteration.

**Comment by Janpot** [comment]
@mnajdova Added a tooltip.



https://user-images.githubusercontent.com/2109932/236806252-442b639e-3c0b-49d6-a432-cd7b5634437a.mov

File: docs/schemas/v1/definitions.json (hunk 1 above)
Comment (Janpot) [reply docs/schemas/v1/definitions.json:83]: It has 
```json
                   "enum": [
                      "query",
                      "mutation"
                    ],
```
I can improve this in the docs

**Pull request merged**